plugins/modules/purefa_snap.py

```diff
@@ -159,243 +167,389 @@
+                == 200
+            )
+        else:
+            return bool(
+                array.get_volume_snapshots(names=[snapname], destroyed=True).status_code
+                == 200
             )
-        except Exception:
-            return False
 
 
 def get_snapshot(module, array):
     """Return True if snapshot exists, False otherwise"""
-    try:
-        snapname = module.params["name"] + "." + module.params["suffix"]
-        name = module.params["name"]
-        if len(name.split(":")) == 2:
-            # API 1.x raises exception if name is a remote snap
-            name = module.params["name"] + "*"
-        for snaps in array.get_volume(name, snap=True, pending=False):
-            if snaps["name"] == snapname:
-                return True
-    except Exception:
-        return False
-
-
-def create_snapshot(module, array, arrayv6):
+    api_version = array.get_rest_version()
+    snapname = module.params["name"] + "." + module.params["suffix"]
+    if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+        return bool(
+            array.get_volume_snapshots(
+                names=[snapname],
+                destroyed=False,
+                context_names=[module.params["context"]],
+            ).status_code
+            == 200
+        )
+    else:
+        return bool(
+            array.get_volume_snapshots(names=[snapname], destroyed=False).status_code
+            == 200
+        )
+
+
+def create_snapshot(module, array):
     """Create Snapshot"""
     changed = False
-    api_version = array._list_available_rest_versions()
+    api_version = array.get_rest_version()
     if module.params["offload"]:
-        if SNAPSHOT_SUFFIX_API not in api_version:
+        if LooseVersion(SNAPSHOT_SUFFIX_API) > LooseVersion(api_version):
             module.params["suffix"] = None
         changed = True
         if not module.check_mode:
-            res = arrayv6.post_remote_volume_snapshots(
-                source_names=[module.params["name"]],
-                on=module.params["offload"],
-                remote_volume_snapshot=flasharray.RemoteVolumeSnapshotPost(
-                    suffix=module.params["suffix"]
-                ),
-            )
+            if LooseVersion(CONTEXT_API_VERSION) <= LooseVersion(api_version):
+                res = array.post_remote_volume_snapshots(
+                    source_names=[module.params["name"]],
+                    context_names=[module.params["context"]],
+                    on=module.params["offload"],
+                    remote_volume_snapshot=RemoteVolumeSnapshotPost(
+                        suffix=module.params["suffix"]
+                    ),
+                )
+            else:
+                res = array.post_remote_volume_snapshots(
+                    source_names=[module.params["name"]],
+                    on=module.params["offload"],
+                    remote_volume_snapshot=RemoteVolumeSnapshotPost(
```

Comment:
getting unknown keyword error, remote_volume_snapshot not required here.

Reply:
`remote_volume_snapshot` is a required field. I'll check what is going on.

Reply:
This is the correct syntax, I just tried it manually, and `remote_volume_snapshot` is required.
If this errors again can you send the full traceback

Reply:
@sdodsley , might be backward compatibility issue, when we ran in purity version - 6.3.4, it errors out

Reply:
@skeerthivasan I think I figured it out. Please retry